src/Main.elm

```diff
@@ -1,10 +1,829 @@
+   library was one of the main reasons I picked up elm.
+
+   Also relevant, Matt Griffith's talk about the elm-ui library:
+       https://www.youtube.com/watch?v=Ie-gqwSHQr0
+-}
+
+
+view : Model -> Html.Html Msg
+view model =
+    let
+        content =
+            case model of
+                Loading loadingModel ->
+                    viewLoading loadingModel
+
+                Loaded loadedModel ->
+                    viewDashboard loadedModel
+
+                JsonError err ->
+                    viewJsonError err
+    in
+    E.layout
+        -- this is the root of the page
+        [ EFont.family
+            -- we set a font family that applies globally
+            [ EFont.typeface Palette.font0
+            ]
+        , EFont.size Palette.fontSize0 -- and a size
+        , EBackground.color Palette.white -- and the background color of the whole page
+        ]
+        -- we split the page into a column of the content and
+        -- the color palette at the bottom
+        <| E.column
+            [ E.width E.fill
+            , E.height E.fill
+            ]
+            [ E.el [ E.paddingXY 0 80, E.centerX, E.height E.fill ] content
+            , E.row
+                [ E.width E.fill
+                , E.alignBottom
+                ]
+                <| List.map viewColor colors
+            ]
+
+
+sortButton : Data.Audience.AudienceType -> Data.Audience.AudienceType -> E.Element Msg
+sortButton audienceType audienceTypeToSortBy =
+    let
+        -- the "label" is a row of the icon and the text
+        label =
+            E.row
+                [ E.spacing 10
+                , E.centerX
+                , E.centerY
+                ]
+                [ E.el
+                    [ E.width <| E.px 18
+                    , E.height <| E.px 18
+                    ]
+                    <| E.html <| Data.Audience.audienceTypeToIcon audienceType
+                , E.text <| Data.Audience.audienceTypeToString audienceType
+                ]
+    in
+    EInput.button
+        [ EBackground.color <|
+            case audienceTypeToSortBy == audienceType of
+                True ->
+                    Palette.color3
+
+                False ->
+                    E.rgba 0 0 0 0
+        , E.width <| E.px 110
+        , E.height <| E.px 40
+        , EBorder.rounded 5
+        , EBorder.width 1
+        , EBorder.color <| Palette.color3
+        ]
+        { label = label
+        , onPress = Just (SortBy audienceType)
+        }
+
+
+
+-- the panel on the right hand side of the dashboard
+
+
+viewSortingPanel : Data.Audience.AudienceType -> E.Element Msg
+viewSortingPanel currentlySortBy =
+    let
+        buttons =
+            List.map
+                (\btn -> btn currentlySortBy)
+                [ sortButton Data.Audience.Authored
+                , sortButton Data.Audience.Curated
+                , sortButton Data.Audience.Shared
+                ]
```

Comment:
```suggestion
            List.map
                (\btn -> sortButton btn currentlySortBy)
                [ Data.Audience.Authored
                , Data.Audience.Curated
                , Data.Audience.Shared
                ]
```
Or you can go with point free if you swap arguments for `sortButton` function.

Reply:
oh you're right!! yeah I missed that one